tests/aws-cpp-sdk-s3control-integration-tests/S3ControlTest.cpp

```diff
@@ -727,4 +727,25 @@ namespace
 
         CleanUpMRAPTest(bucketName, multiRegions, accessPointName);
     }
+
+
+
+TEST_F(S3ControlTest, TestRemoveHostPrefix)
+{
+    ClientConfiguration config;
+    config.region = Aws::Region::US_EAST_1;
+    auto s3controlClient = S3ControlClient(config);
+    Aws::String bucket = "test-ap-bucket--use2-az1--x-s3";
+    Aws::String accessPointName = "test-ap--use2-az1--xa-s3";
+    auto request = CreateAccessPointRequest{}
+    .WithName(accessPointName)
+    .WithBucket(bucket)
+    .WithAccountId(m_accountId)
+    .WithBucketAccountId(m_accountId);
+    auto resolveEndpointOutcome = s3controlClient.accessEndpointProvider()->ResolveEndpoint(request.GetEndpointContextParams());
```

Comment:
this doesnt actually test the change you made, it tests the endpoint provider, which worked correctly before. [You're actually changing client source](https://github.com/aws/aws-sdk-cpp/pull/3421/files#diff-7b79c9c5623c1367bbb15a9e243bc6be7518bf34dadfe8e5fe206291c552c2c7L463-L464), so a adequate test would be calling `s3controlClient.CreateAccessPoint`